config/initializers/content_security_policy.rb

```diff
@@ -5,12 +5,22 @@
 # https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy
 
 Rails.application.config.content_security_policy do |policy|
+  # Attributes clarification
+  # :self, only allow resources from current origin
+  # :none, won't allow loading of any resources
+  #
+  # :https, requires secure http to load resources
+  #
+  # :unsafe_eval, allows execution of eval statements
+  # :unsafe_inline, allows inline resources usage
+
   policy.default_src :self, :https
-  policy.font_src    :self, :https, :data
-  policy.img_src     :self, :https, :data
+  policy.font_src    :self, :https
+  policy.img_src     :self, :https
   policy.object_src  :none
-  policy.script_src  :self, :https, :unsafe_eval, :unsafe_inline, "http://localhost:4200"
-  policy.style_src   :unsafe_inline, :self, :https, "http://localhost:4200"
+  policy.script_src  :self, :https, :unsafe_eval, "http://localhost:4200"
+  policy.style_src   :self, :https, :unsafe_inline, "http://localhost:4200"
```

Comment:
Gleichermassen bei `unsafe_inline` für Styles: Dies erlaubt `style=""` Styles in der Seite, was ein [Security-Risiko](https://www.youtube.com/watch?v=eb3suf4REyI) sein kann.

Anstatt die CSP zu lockern, könnten wir herausfinden wo in der Seite überall solche inline `style="..."` Attribute benutzt werden, und diese alle durch fixe Styles mit CSS Klassen ersetzen?

Reply:
Und auch hier würde ich `http://localhost:4200` nur im Dev Modus behalten.

Reply:
Mein Browser gibt mir zwar keine Fehler mehr, aber immer noch folgende Warnung:
```
Content Security Policy: Ignoring “'unsafe-inline'” within script-src or style-src: nonce-source or hash-source specified
```

Wenn ich die entsprechende [Firefox-Feature-Beschreibung](https://bugzilla.mozilla.org/show_bug.cgi?id=1004703) richtig verstehe, wird `:unsafe_inline` für Skripts und auch für Styles ignoriert, sobald eine Nonce da ist. Du hast zwar keine Nonce explizit spezifiziert, aber Rails macht anscheinend trotzdem mal auf Vorrat eine:
```
... script-src 'self' https: 'unsafe-eval' 'unsafe-inline' http://localhost:4200 'nonce-TaGl5Do3J65h68dd+HyUcg=='; ...
```
Die Idee ist, dass Seiten, die von `unsafe_inline` auf eine Nonce migrieren wollen, einfach mal eine Nonce einfügen können, und die Seite aber auf alten Browsern die noch keine Noncen unterstützen weiterhin funktioniert.

**Lange Rede kurzer Sinn**: Wenn wir davon ausgehen dass alle Browser Noncen unterstützen (inzwischen ist IE der einzige der das [nicht unterstützte](https://caniuse.com/contentsecuritypolicy2)), kannst du `:unsafe_inline` genausogut entfernen, es wird sowieso ignoriert.